src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs

```diff
@@ -218,15 +214,23 @@ private SynthesizedReadOnlyListTypeSymbol(SourceModuleSymbol containingModule, s
             var iCollectionT = compilation.GetSpecialType(SpecialType.System_Collections_Generic_ICollection_T).Construct(typeArgs);
             var iListT = compilation.GetSpecialType(SpecialType.System_Collections_Generic_IList_T).Construct(typeArgs);
 
-            _interfaces = ImmutableArray.Create(
-                iEnumerable,
-                iCollection,
-                iList,
-                iEnumerableT,
-                iReadOnlyCollectionT,
-                iReadOnlyListT,
-                iCollectionT,
-                iListT);
+            _interfaces = iReadOnlyCollectionT.IsErrorType() || iReadOnlyListT.IsErrorType()
```

Comment:
>iReadOnlyCollectionT.IsErrorType() || iReadOnlyListT.IsErrorType() [](http://example.com/codeflow?start=26&length=66)

Should we check `compilation.IsTypeMissing(specialType)` for these cases rather than assuming an error type is only generated for missing types?

Reply:
That sounds better, thanks!